torch/_export/__init__.py

```diff
@@ -366,7 +366,7 @@ def optimized(*args, **kwargs):
         out_spec = pytree.treespec_loads(call_spec[1])
         flat_inputs = pytree.tree_flatten((args, reorder_kwargs(kwargs, in_spec)))[0]
         flat_inputs = [x for x in flat_inputs if isinstance(x, torch.Tensor)]
-        flat_outputs = runner.run(flat_inputs)  # type: ignore[attr-defined]
+        flat_outputs = runner.run(flat_inputs, None)  # type: ignore[attr-defined]
```

Comment:
Since we've changed the runner API from `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs)` to `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs, void* stream_handle = nullptr)`, Here wiil be error like:
```
ERROR: test_amp_fallback_random_cpu (__main__.AOTInductorTestABICompatibleCpu)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/xinanlin/xinanlin/pytorch/torch/testing/_internal/common_utils.py", line 3062, in wrapper
    method(*args, **kwargs)
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_torchinductor.py", line 11757, in new_test
    return value(self)
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_aot_inductor.py", line 524, in test_amp_fallback_random
    self.check_model(fn, example_inputs)
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_aot_inductor.py", line 138, in check_model
    actual = AOTIRunnerUtil.run(
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_aot_inductor_utils.py", line 132, in run
    return optimized(*example_inputs)
  File "/home/xinanlin/xinanlin/pytorch/torch/_export/__init__.py", line 369, in optimized
    flat_outputs = runner.run(flat_inputs)  # type: ignore[attr-defined]
TypeError: run(): incompatible function arguments. The following argument types are supported:
    1. (self: torch._C._aoti.AOTIModelContainerRunnerCpu, arg0: list[torch.Tensor], arg1: capsule) -> list[torch.Tensor]

Invoked with: <torch._C._aoti.AOTIModelContainerRunnerCpu object at 0x7fc168a5fdf0>, [tensor([[-0.1117, -0.4966,  0.1631, -0.8817,  0.0539,  0.6684, -0.0597, -0.4675
,
```
The Pybind Python API `run` does not know how to process the default void* stream_handle = nullptr, So we explicitly assign a None for it.

Reply:
@desertfire  @EikanWang , After changed the API as you suggested, we also get the following error, which use `runner.run(input)` but the changed Cpp API `std::vector<at::Tensor> run(const std::vector<at::Tensor>& inputs, void* stream_handle = nullptr)` can not handle this. The Pybind Python API run does not know how to process the default void* stream_handle = nullptr, So we need to explicitly assign a None for it. This is a real API break change now.
I think my previous design maybe better?
======================================================================
ERROR: test_non_tensor_input_cpu_with_stack_allocation (__main__.AOTInductorTestABICompatibleCpuWithStackAllocation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/xinanlin/xinanlin/pytorch/torch/testing/_internal/common_utils.py", line 3095, in wrapper
    method(*args, **kwargs)
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_torchinductor.py", line 11822, in new_test
    return value(self)
  File "/home/xinanlin/xinanlin/pytorch/test/inductor/test_aot_inductor.py", line 1570, in test_non_tensor_input
    res = kernel_runner.run([a, b])
TypeError: run(): incompatible function arguments. The following argument types are supported:
    1. (self: torch._C._aoti.AOTIModelContainerRunnerCpu, arg0: list[torch.Tensor], arg1: capsule) -> list[torch.Tensor]

Invoked with: <torch._C._aoti.AOTIModelContainerRunnerCpu object at 0x7fcf7182ce30>, [tensor([ 0.0461,  0.4024, -1.0115,  0.2167, -0.6123,  0.5036,  0.2310,  0.6931,
        -0.2669,  2.1785]), tensor([ 0.1021, -0.2590, -0.1549, -1.3706, -0.1319,  0.8848, -0.2611,  0.6104,
        -0.0098, -1.4473])]

To execute this test, run the following from the base repo dir:
    python test/inductor/test_aot_inductor_arrayref.py AOTInductorTestABICompatibleCpuWithStackAllocation.test_non_tensor_input_cpu_with_stack_allocation

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

Reply:
The cpp-side AOTI runner API should be the main integration point for AOTI, since after all it is about deploying models in cpp. I consider these pybind interfaces as internal APIs, so I am fine with breaking them instead of breaking the main runner API in cpp. 

The reason why we had these pybind interfaces in the first place is to facilitate Python side runner interface, previously `torch._export.aot_load` and more recently `torch._inductor.aoti_load_package`. You may need to update their implementation because of the issue you are mentioning here.

Reply:
> The cpp-side AOTI runner API should be the main integration point for AOTI, since after all it is about deploying models in cpp. I consider these pybind interfaces as internal APIs, so I am fine with breaking them instead of breaking the main runner API in cpp.
> 
> The reason why we had these pybind interfaces in the first place is to facilitate Python side runner interface, previously `torch._export.aot_load` and more recently `torch._inductor.aoti_load_package`. You may need to update their implementation because of the issue you are mentioning here.

@desertfire thanks for your time, I've updated the interface to support `void* stream_handle=nullptr`.